plugins/workflow/notification/notification.php

```diff
@@ -0,0 +1,252 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Workflow.Notification
+ *
+ * @copyright   Copyright (C) 2005 - 2020 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Application\CMSApplicationInterface;
+use Joomla\CMS\Component\ComponentHelper;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Form\Form;
+use Joomla\CMS\Language\Language;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Object\CMSObject;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Table\TableInterface;
+use Joomla\CMS\User\User;
+use Joomla\CMS\Workflow\Workflow;
+use Joomla\Component\Content\Administrator\Table\ArticleTable;
+use Joomla\Database\ParameterType;
+use Joomla\Utilities\ArrayHelper;
+
+/**
+ * Workflow Notification Plugin
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class PlgWorkflowNotification extends CMSPlugin
+{
+	/**
+	 * Load the language file on instantiation.
+	 *
+	 * @var    boolean
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $autoloadLanguage = true;
+
+	/**
+	 * Loads the CMS Application for direct access
+	 *
+	 * @var   CMSApplicationInterface
+	 * @since __DEPLOY_VERSION__
+	 */
+	protected $app;
+
+	/**
+	 * Database object.
+	 *
+	 * @var    JDatabaseDriver
+	 * @since  3.9.0
+	 */
+	protected $db;
+
+	/**
+	 * The form event.
+	 *
+	 * @param   Form      $form  The form
+	 * @param   stdClass  $data  The data
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onContentPrepareForm(Form $form, $data)
+	{
+		$context = $form->getName();
+
+		// Extend the transition form
+		if ($context !== 'com_workflow.transition')
+		{
+			return;
+		}
+
+		return $this->enhanceTransitionForm($form, $data);
+	}
+
+	/**
+	 * Add different parameter options to the transition view, we need when executing the transition
+	 *
+	 * @param   Form      $form  The form
+	 * @param   stdClass  $data  The data
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function enhanceTransitionForm(Form $form, $data)
+	{
+		Form::addFormPath(__DIR__ . '/forms');
+
+		$form->loadFile('workflow_notification');
+
+		return true;
+	}
+
+	/**
+	 * Send a Notification to defined users a transion is performed
+	 *
+	 * @param   string   $context  The context for the content passed to the plugin.
+	 * @param   array    $pks      A list of primary key ids of the content that has changed stage.
+	 * @param   object  $data    Object containing data about the transition
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onWorkflowAfterTransition($context, $pks, $data)
+	{
+		// Check if send-mail is active
```

Comment:
At the beginning we need the check, if the workflow is active + the workflow is supported at all:
https://github.com/bembelimen/joomla-cms/blob/workflow-v3/plugins/workflow/publishing/publishing.php#L408-L419
(nut not the check from line 420). Otherwise we're not sure, that the method ```getModelName``` is supported.

Reply:
check for isWorkflowActive added, check for method getItem added